src/rpc/mining.cpp

```diff
@@ -100,6 +100,88 @@ static UniValue getnetworkhashps(const JSONRPCRequest& request)
     return GetNetworkHashPS(!request.params[0].isNull() ? request.params[0].get_int() : 120, !request.params[1].isNull() ? request.params[1].get_int() : -1);
 }
 
+static bool GenerateBlock(const CTxMemPool& mempool, const CScript& coinbase_script, const std::vector<CTransactionRef>* preselected_txs, uint64_t& max_tries, unsigned int& extra_nonce, uint256& block_hash)
+{
+    block_hash.SetNull();
+
+    CBlock block;
+    CChainParams chainparams(Params());
+
+    if (!preselected_txs) {
+        std::unique_ptr<CBlockTemplate> pblocktemplate(BlockAssembler(mempool, chainparams).CreateNewBlock(coinbase_script));
```

Comment:
Why is this code moved into this function? It could stay at the original place and here you just pass in a `shared_ptr` of the resulting block?